application-mocca-calendar-api/src/main/java/org/xwiki/contrib/moccacalendar/internal/importJob/CalendarEventImporter.java

```diff
@@ -0,0 +1,191 @@
+        eventObj.set(EventConstants.PROPERTY_ALLDAY_NAME, allDay, wikiContext);
+        if (allDay == 1) {
+            LocalDateTime end = component.getEnd().toInstant().atZone(ZoneOffset.UTC).toLocalDateTime().minusDays(1);
+            eventObj.set(EventConstants.PROPERTY_ENDDATE_NAME, Date.from(end.atZone(ZoneOffset.UTC).toInstant()),
+                wikiContext);
+        } else {
+            eventObj.set(EventConstants.PROPERTY_ENDDATE_NAME, component.getEnd(), wikiContext);
+        }
+        eventObj.set(EventConstants.PROPERTY_STARTDATE_NAME, component.getStart(), wikiContext);
+
+        int recurrenceValue = component.isRecurrent();
+        eventObj.set(EventConstants.PROPERTY_RECURRENT_NAME, recurrenceValue, wikiContext);
+        if (recurrenceValue == 1) {
+            processRecurrence(eventDoc, component);
+        }
+    }
+
+    /**
+     * Checks if a {@link XWikiDocument} with the given name exists at the given parent reference or is to be processed
+     * and generate a unique name for it.
+     *
+     * @param eventName name of the event to be added.
+     * @param parentRef reference to the parent of the event.
+     * @param eventDocuments {@link List} with the events document that are to be processed and created.
+     * @return a {@link XWikiDocument} with the title the given event name and a unique identifier.
+     * @throws XWikiException if there are any exceptions while getting the {@link XWikiDocument} for the event.
+     */
+    public XWikiDocument getUniqueEventName(String eventName, String parentRef, List<XWikiDocument> eventDocuments)
+        throws XWikiException
+    {
+        XWikiContext wikiContext = this.wikiContextProvider.get();
+
+        DocumentReference parentDocRef = documentReferenceResolver.resolve(parentRef);
+        SpaceReference documentSpaceRef = new SpaceReference(eventName, parentDocRef.getLastSpaceReference());
+        DocumentReference eventRef = new DocumentReference(NEW_EVENT_HOME, documentSpaceRef);
+
+        XWikiDocument eventDoc = wikiContext.getWiki().getDocument(eventRef, wikiContext);
+
+        if (!eventDoc.isNew() || sameNameEventExists(eventRef, eventDocuments)) {
+            String newEventName = eventName + "_" + System.nanoTime();
+            SpaceReference newDocumentSpaceRef = new SpaceReference(newEventName, parentDocRef.getLastSpaceReference());
+            DocumentReference newEventRef = new DocumentReference(NEW_EVENT_HOME, newDocumentSpaceRef);
+            eventDoc = new XWikiDocument(newEventRef);
+        }
+        eventDoc.setTitle(eventName);
+        eventDoc.setParentReference(parentDocRef);
+        return eventDoc;
+    }
+
+    private void addConvertedPropertyToObject(String content, BaseObject eventObj, String syntax, String property)
+    {
+        String cleanContent = Jsoup.clean(content, Safelist.basic());
+        String convertedContent = htmlConverter.fromHTML(cleanContent, syntax);
```

Comment:
Could you add a comment explaining why this is needed? I feel it's not that intuitive to understand why. Even we needed some time in the meeting to understand what could be best